### ISSUE TITLE
Comments not different enough.

### ISSUE THREAD

**Issue by Arithmeticae** [arcticicestudio/nord-visual-studio-code#14]
I know that much of a user's preference when it comes to a theme can be driven by small changes in the settings of their display. So if others aren't experiencing the same, I apologize.

I find that on my display, and many of my coworkers' displays, the different between comments and the background color is not substantial which causes the comments to be difficult to read when the lighting in the office isn't perfect.

I know that it's not as simple as lightening up the comment (and token) color as that might interfere with the line highlight color but I hope you can find a solution that would make it just a little easier to read.

**Comment by arcticicestudio** [comment]
It is not easy to improve this when the theme engine is limited to defined JSON attributes only. In other port projects it is possible to improve this by using advanced theming features like Sass/LESSCSS/CSS.
It would be nice if VS Code supports custom theme settings per JavaScript classes which would allow to implement a setting to increase percentage contrast of the comment color like the one in the [Nord Atom UI](https://github.com/arcticicestudio/nord-atom-syntax/pull/47).

**Comment by LeviticusMB** [comment]
Really like this theme, but the comment color forces me to switch back to another theme after a while. 😢

**Comment by ajitid** [comment]
Same here, only issue with this theme.

**Comment by arcticicestudio** [comment]
Customize a theme via settings and impement a API for theme authors are one of the most requested VS Code features as you can see in [Microsoft/vscode #11556](https://github.com/Microsoft/vscode/issues/11556) and [Microsoft/vscode #459](https://github.com/Microsoft/vscode/issues/459).

It seems like the team is going to refactor the current JSON format again as seen in the proof on concept PR [Microsoft/vscode #21981](https://github.com/Microsoft/vscode/pull/21981).
![](https://cloud.githubusercontent.com/assets/2957173/23581026/14c245ec-010d-11e7-904b-f93fdf258682.gif)
![](https://cloud.githubusercontent.com/assets/2957173/23581030/1f0acd26-010d-11e7-8302-9f6cbb8e682c.gif)

I hope they implement it[ the great way like the Atom team did](https://atom.io/docs/api/latest) which may allow me to implement accessbility settings or at least users can adjust themes to their personal needs.

**Comment by Lott** [comment]
Comment color is my only issue with this theme, everything else is fantastic.  While certainly not the best way, you can also change the comment color in nord.json in 0.2.0 Search for "Comment" (with the quotes) and change the foreground value (~line 150).  I use #717C96

**Comment by ajitid** [comment]
Can we get a temporary fix for this?

**Comment by wprater** [comment]
> Can we get a temporary fix for this?

My fix is to switch to another theme every once in a while .. :|

**Comment by arcticicestudio** [comment]
Sorry for my late reply, haven't got the time to work on any private projects the last months.
> Can we get a temporary fix for this?

Honestly there is nothing that needs to be "fixed", it's a personal preference. The color has been chosen to keep the focus of the user on the code by "fading out" comments/docs. The color works find if the settings of the monitor are correct regarding the contrast and brightness based on environment.

I've got a lot of requests in other ports projects too, but unfortunately VS Code is still not able to provide custom settings per extension compared to other editors which allowed me to implement a "accessibility" setting where the user can increase the contrast using percentage values.

Anyway, I've added [this card](https://github.com/arcticicestudio/nord-visual-studio-code/projects/1#card-3496031) regarding [Microsoft/vscode #27894](https://github.com/Microsoft/vscode/issues/27894) which finally allows users to override syntax (token) colors using a `editor.tokenColorsCustomizations` setting object.
This way everyone how'd like to increase the contrast of the comment color can do it on it's own. I hope they implement it extension-scoped so this override rule doesn't effect other themes.

**Comment by arcticicestudio** [comment]
I just want to clarify that I don't want to blame anyone for having bad monitor settings or general setup, just the fact that the used color is adjusted to the theme.
**I understand that it might be too dark (low contrast) for some users which are familiar with brighter comment colors, but like I said before that this is a personal preference.**

I'd really like to solve this to make this theme usable for all kind of personal preferences, but it relies to the VS Code development team to implement a way for theme authors to provide customization settings or at least allow to override theme settings.

There is a PR which implements the new token color override feature which I'll keep track of.
Regarding to this  official VS Code team member [comment](https://github.com/Microsoft/vscode/pull/29393#issuecomment-310971661) in the PR they plan to push it to the July insiders version.

The [backlog card](https://github.com/arcticicestudio/nord-visual-studio-code/projects/2#card-3496506) now also links to the PR which I'll track.

**Comment by arcticicestudio** [comment]
Microsoft/vscode#29393 has been merged and can be tested with the latest Insiders build!
This means you can now override token colors defined by the syntax theme and customize it to fit your needs.

To change the color of all comments for the Nord theme you can add this to your `settings.json`...
```json
  "editor.tokenColorCustomizations": {
    "textMateRules": [
      {
        "name": "Comment",
        "scope": "comment",
        "settings": {
          "foreground": "#4C566A"
        }
      }
    ]
  }
```

...and replace the current comment color `#4C566A` with your preferred one.

This is a reference table if you'd like to use the same increased contrast values as provided by the [Nord Atom Syntax accessibility custom comment contrast theme setting feature](https://github.com/arcticicestudio/nord-atom-syntax/pull/47) which are calculated using the LESSCSS [`lighten`](http://lesscss.org/functions/#color-operations-lighten) function.

| Increased by | Calculated value |
| --- | --- |
| 1% | `#4e586d` |
| 2% | `#505b70` |
| 3% | `#525d73` |
| 4% | `#556076` |
| 5% | `#576279` |
| 6% | `#59647c` |
| 7% | `#5b677f` |
| 8% | `#5d6982` |
| 9% | `#5f6c85` |
| 10% | `#616e88` |
| 11% | `#63718b` |
| 12% | `#66738e` |
| 13% | `#687591` |
| 14% | `#6a7894` |
| 15% | `#6d7a96` |
| 16% | `#6f7d98` |
| 17% | `#72809a` |
| 18% | `#75829c` |
| 19% | `#78859e` |
| 20% | `#7b88a1` |

**Comment by peterdanis** [comment]
Just a small update, Microsoft/vscode#29393 is now available also in stable channel, release 1.15.

**Comment by jpeterson92** [comment]
Making this change in my own version since release 1.15 has been a marked increase in usability for me! Thanks for a great color scheme.

**Comment by ajitid** [comment]
I still wish for 5% `#576279` to be the default one. Everyone's issue with this theme is just this.

**Comment by arcticicestudio** [comment]
@ajitid I've researched the API documentations and maybe there's a way to implement a theme settings to provide a way for users to set a percent value the comment colors will be increased by.

Changing the default color would affect all Nord port projects since I've to adapt the change to all of them. Maybe I'll start a vote poll for it and if most users would like to see the comment color to be increased by default I'll adapt it.

**Comment by bsvipas** [comment]
@arcticicestudio Lighten by 15% is perfect!

**Comment by octref** [comment]
@arcticicestudio https://github.com/equinusocio/vsc-material-theme has a setting you can access by Cmd + Shift + P -> Material Theme: Settings. Users can use it to change color accents and color variants. Maybe you can also do something similar.

**Comment by arcticicestudio** [comment]
> I've researched the API documentations and maybe there's a way to implement a theme settings to provide a way for users to set a percent value the comment colors will be increased by.

@octref Thanks for the link, the Material Theme uses the same way of the API I've also found, it's a bit "hacky" and I think not meant to be used this way, but until they extend the theme API to provide this functionality it might be the only way to implement such a feature.

**Comment by arcticicestudio** [comment]
I've increased the comment color by `5%` up to `8%` **locally** for all port projects I use every day and finally came to the decision that it should be fine to increase the current comment color by default.

I'll open a vote thread in the Nord repository within the next weeks to get some feedback from all users of all port projects and to decide the value which should be between `5` and `8` percent.

**Comment by fxn** [comment]
This reminds me of an Emacs function I wrote many years ago to darken/light Javadoc, also fold/unfold all Javadoc in a file, because Javadoc tended to be very verbose.

I believe the motivation for the default is (was?) precisely something in that line, to deemphasize docs and comments and let the code get the focus. Problem is sometimes you need to read the comments, sometimes you need to read the docs, and sometimes you are the one writing comments and docs! You need to see them clearly in those use cases.

Thanks for the table with %s above!

**Comment by fxn** [comment]
Note that starting with version 1.20 of VS Code, the customization can be scoped to the theme:

```json
"editor.tokenColorCustomizations": {
  "[Nord]": {
    "textMateRules": [
      {
        "name": "Comment",
        "scope": "comment",
        "settings": {
          "foreground": "#5d6982"
        }
      }
    ]
  }
}
```

**Comment by ChrisFerrell** [comment]
@arcticicestudio Thanks for the settings.json instructions and the color table override.  Nord is perfect now.  Love the theme

**Comment by guyinpv** [comment]
I didn't have to use the "textMateRules" thing any more. 
I could change the comment color just with this:
```
"editor.tokenColorCustomizations": {
        "[Nord]": {
            "comments": "#717c96"
        }
    }
```

**Comment by octref** [comment]
I'm not sure if I should open another issue, but one big problem for me is in diff view I literally can't read the comments:

![image](https://user-images.githubusercontent.com/4033249/55448789-0df65b00-557e-11e9-939a-82565d6cafc8.png)

**Comment by arcticicestudio** [comment]
@octref There's #105 that I'd like to merge before deploying the next version, but I'm currently also stuck at solving this problem. If there would be theme scopes that would allow to explicitly style elements that are highlighted in a diff-view I could simply adjust all colors, but there's no good style that creates a nice coloring for both diff and normal views.

By the way, the long-time problem with comments has been solved in #118 (#117) and will also be included in the next release :tada:

**Comment by octref** [comment]
I opened https://github.com/Microsoft/vscode/issues/71663.

**Comment by arcticicestudio** [comment]
@octref Thanks :green_heart
This will definitely help a lot of theme authors to fix this contrast issue.

**Comment by arcticicestudio** [comment]
Alright, its finally time to close this long.time issue :tada: 

PR #118 (implementing https://github.com/arcticicestudio/nord/issues/94 through #117) has been merged and version [0.8.0][] has been release and deployed to the VS Code Marketplace :rocket:

The comment's color brightness (`nord3`) is now increased by 10% from a lightness level of ~35% to ~45%.

Feel free to let me know what you think about this change!
➜ Please see https://github.com/arcticicestudio/nord/issues/94 for all details about this design change decision.

#### Before

![gh-146-before](https://user-images.githubusercontent.com/7836623/54902736-76886c80-4eda-11e9-86cd-dfc935aff5e3.png)
![gh-145-before-2](https://user-images.githubusercontent.com/7836623/54902735-76886c80-4eda-11e9-9aa0-41ded647bdb2.png)

#### After

![gh-146-preview](https://user-images.githubusercontent.com/7836623/54902766-856f1f00-4eda-11e9-897a-9b0971586a0b.png)
![gh-145-preview-2](https://user-images.githubusercontent.com/7836623/54902765-856f1f00-4eda-11e9-9d09-50c89faece43.png)

[0.8.0]: https://github.com/arcticicestudio/nord-visual-studio-code/releases/tag/v0.8.0
